streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java

```diff
@@ -720,6 +729,86 @@ public static void waitUntilMetadataIsPropagated(final List<KafkaServer> servers
 
     }
 
+    /**
+     * Starts the given {@link KafkaStreams} instances and waits for all of them to reach the
+     * {@link State#RUNNING} state at the same time. Note that states may change between the time
+     * that this method returns and the calling function executes its next statement.
+     *
+     * @param streamsList the list of streams instances to run.
+     * @param timeout the time to wait for the streams to all be in @{link State#RUNNING} state.
+     */
+    public static void startApplicationAndWaitUntilRunning(final List<KafkaStreams> streamsList,
```

Comment:
We recently merge a fix for EosIntegrationTest that also starts two instances -- can we update this test to use this new method in this PR? https://github.com/apache/kafka/blob/trunk/streams/src/test/java/org/apache/kafka/streams/integration/EosIntegrationTest.java#L467-L468

Also, the other PR introduced `StreamsTestUtils#startKafkaStreamsAndWaitForRunningState(...)` that takes a long as timeout -- to align with this PR, we could change it to `Duration` as well.

We can also do a follow up PR if you don't want to block this PR by both.

Reply:
Do we need both startKafkaStreamsAndWaitForRunningState and this? Can we just consolidate them into a single function?